src/me/drton/jmavsim/SimpleSensors.java

```diff
@@ -179,8 +182,16 @@ public void update(long t) {
             GNSSReport gpsCurrent = new GNSSReport();
             eph = (float)ephFilter.filter(ephLow);
             epv = (float)epvFilter.filter(epvLow);
+
+            // add noise (random walk)
+            Vector3d zeroVector = new Vector3d();
+            Vector3d whiteNoise = addZeroMeanNoise(zeroVector, gpsNoiseStdDev);
+            randomWalkNoise.add(whiteNoise);
+            Vector3d noisedPos = new Vector3d(object.getPosition());
+            noisedPos.add(randomWalkNoise);
+            setGlobalPosition(noisedPos);
```

Comment:
The computation of the noise in here is correct, but it needs to be added inside `setGlobalPosition()` (add `randomWalkNoise` to `pos`). Because `setGlobalPosition` can be called independently and/or more often than this here.

Reply:
Thank you very much for your advice @bkueng  :-)

I will modify the gps noise considering dt.

Thanks.